src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -0,0 +1,362 @@
+                  std::vector<float>& mel_data,
+                  const size_t feature_size,
+                  const size_t nb_max_frames) {
+    OPENVINO_ASSERT(mel_data.size() == feature_size * nb_max_frames,
+                    "Mel spectrogram required size: ",
+                    feature_size,
+                    " * ",
+                    nb_max_frames,
+                    ". Actual size: ",
+                    mel_data.size(),
+                    ".");
+    copy_to_tensor(mel_data, request.get_tensor("input_features"));
+    request.infer();
+    return request.get_tensor("last_hidden_state");
+}
+
+// FIXME: Duplicate from llm_pipeline_static.cpp - need to reuse instead of copy-paste
+ov::Tensor make_tensor_slice(ov::Tensor tensor, size_t dim, size_t start_pos, size_t end_pos) {
+    ov::Shape start_shape(std::vector<size_t>(tensor.get_shape().size(), 0u));
+    start_shape[dim] = start_pos;
+    ov::Shape end_shape = tensor.get_shape();
+    end_shape[dim] = end_pos;
+    return ov::Tensor(tensor, start_shape, end_shape);
+}
+
+void set_cross_attn_key_value(ov::InferRequest& source, ov::InferRequest& dest) {
+    // NB: Source outputs:
+    // present_key_values.0.encoder.key
+    // present_key_values.0.encoder.value
+
+    // NB: Dest inputs:
+    // past_key_values.0.encoder.key
+    // past_key_values.0.encoder.value
+
+    for (auto& source_output : source.get_compiled_model().outputs()) {
+        std::string source_output_name = source_output.get_any_name();
+        if (source_output_name.find("encoder") == std::string::npos) {
+            continue;
+        }
+        std::string with_past_input_name =
+            std::regex_replace(source_output_name, std::regex("present"), "past");
+        dest.set_tensor(with_past_input_name, source.get_tensor(source_output_name));
+    }
+}
+
+void update_past_key_value(ov::InferRequest& source,
+                           ov::InferRequest& dest,
+                           const size_t kv_pos = 0u) {
+    // NB: Source outputs:
+    // present_key_values.0.decoder.key
+    // present_key_values.0.decoder.value
+
+    // NB: Dest inputs:
+    // past_key_values.0.decoder.key
+    // past_key_values.0.decoder.value
+
+    for (auto& source_output : source.get_compiled_model().outputs()) {
+        std::string source_output_name = source_output.get_any_name();
+        if (source_output_name.find("decoder") == std::string::npos) {
+            continue;
+        }
+
+        std::string with_past_input_name =
+            std::regex_replace(source_output_name, std::regex("present"), "past");
+
+        auto src_kv_tensor = source.get_tensor(source_output_name);
+        auto dst_kv_tensor = dest.get_tensor(with_past_input_name);
+        auto kv_size = src_kv_tensor.get_shape()[2];
+        // NB: Copy src_kv_tensor into dst_kv_tensor[:, :, kv_pos:kv_pos+kv_size, :]
+        auto dst_kv_tensor_slice = make_tensor_slice(dst_kv_tensor, 2u, kv_pos, kv_pos + kv_size);
```

Comment:
For the next optimization, considering optimal layout, seq length (kv_size) should be in dim 2 for keys and dim 3 for values.